src/plugins/clips-executive/clips/domain.clp

```diff
@@ -216,263 +633,188 @@
   (modify ?a (param-names ?param-names))
 )
 
-(defrule domain-add-operator-to-top-precondition
-  "If a precondition has no operator but is part of a name that is an operator,
-   add that name as operator to the precondition."
-  ?precond <- (domain-precondition (operator nil) (part-of ?op))
-  (domain-operator (name ?op))
-=>
-  (modify ?precond (operator ?op))
-)
-
-(defrule domain-add-operator-to-nested-precondition
-  "If a precondition does not have an operator but the parent has one, then copy
-   the operator from the parent to the child."
-  ?precond <- (domain-precondition (operator nil) (part-of ?parent))
-  (domain-precondition (name ?parent) (operator ?op&~nil))
-=>
-  (modify ?precond (operator ?op))
-)
-
-(defrule domain-add-operator-to-atomic-precondition
-  "If an atomic precondition does not have an operator but the parent has one,
-   then copy the operator from the parent to the child."
-  ?precond <- (domain-atomic-precondition (operator nil) (part-of ?parent))
-  (domain-precondition (name ?parent) (operator ?op&~nil))
-=>
-  (modify ?precond (operator ?op))
-)
-
 (deffunction remove-precondition
-  "Remove an atomic precondition from its parent and clean up the precondition
-   tree. If the parent is a disjunction with no other disjunct, simplify it to
+  "Remove a sub-formula from its parent and clean up the formula tree.
+   If the parent is a disjunction with no other disjunct, simplify it to
    true by removing it recursively. If it is a negation, remove it recursively.
    If it's a conjunction, only remove the conjunct.
-   If the top-most precondition is removed, replace it by a trivially true one
+   If the top-most formula node is removed, replace it by a trivially true one
    (empty conjunction)."
   (?precond-fact)
```

Comment:
IIUC this will always be a predicate, correct? If so, the names should be adapted accordingly. Also, some conditions below could be simplified (as we know that the parent will be an atomic formula).

Reply:
The way it is used right now yes, but the function was written in such a way that it could remove any subformula. We can either keep it like that or restrict it to predicates.

Reply:
OK makes sense to keep it as it is